src/mono/wasm/runtime/hybrid-globalization/normalization.ts

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { setU16 } from "../memory";
+import { mono_wasm_new_external_root } from "../roots";
+import { conv_string_root } from "../strings";
+import { MonoString, MonoStringRef } from "../types";
+import { Int32Ptr } from "../types/emscripten";
+import { pass_exception_details } from "./common";
+
+export function mono_wasm_is_normalized(exceptionMessage: Int32Ptr, normalizationForm: number, inputStr: MonoStringRef) : number{
+    const inputRoot = mono_wasm_new_external_root<MonoString>(inputStr);
+    try{
+        const jsString = conv_string_root(inputRoot);
+        if (!jsString)
+            throw new Error("Invalid string was received.");
+
+        const normalization = normalization_to_string(normalizationForm);
+        const result = jsString.normalize(normalization);
+        return result === jsString ? 1 : 0;
+    }
+    catch (ex) {
+        pass_exception_details(ex, exceptionMessage);
```

Comment:
what sets the `exceptionMessage` to null if there is no exception. 

Also consider adding `is_exception: Int32Ptr` argument to all methods which could throw. We use that pattern on other places.

Reply:
It will be done, but let's not do it here because it requires changes in all exisitng hg functions.